src/cascade_at/dismod/api/dismod_extractor.py

```diff
@@ -18,18 +19,99 @@ class DismodExtractor(DismodIO):
+        predictions['rate'] = predictions['integrand'].map(PRIMARY_INTEGRANDS_TO_RATES)
+        if location_id is not None:
+            predictions = predictions.loc[predictions.c_location_id == location_id].copy()
+        if sex_id is not None:
+            predictions = predictions.loc[predictions.c_sex_id == sex_id].copy()
+        return predictions
+
+    def gather_draws_for_prior_grid(self, location_id, sex_id, rates, value=True, dage=True, dtime=True):
+        """
+        Takes draws and formats them for a prior grid for values, dage, and dtime.
+        Assumes that age_lower == age_upper and time_lower == time_upper for all
+        data rows. We might not want to do all value, dage, and dtime, so pass False
+        if you want to skip those.
+
+        Args:
+            location_id: (int)
+            sex_id: (int)
+            rates: List[str] list of rates to get the draws for
+            value: (bool) calculate value priors
+            dage: (bool) calculate dage priors
+            dtime: (bool) calculate dtime priors
+
+        Returns:
+            draw_data: (np.ndarray) 3-d array of value draws over age and time for this loc and sex
+            draw_dage: (np.ndarray) 3-d array of draws for dage over age and time for this loc and sex
+            draw_dtime: (np.ndarray) 3-d array of draws for dtime over age and time for this loc and sex
+        """
+        rate_dict = dict()
+        for r in rates:
+            rate_dict[r] = dict()
+
+        df = self.get_predictions(location_id=location_id, sex_id=sex_id)
+
+        assert (df.age_lower.values == df.age_upper.values).all()
+        assert (df.time_lower.values == df.time_upper.values).all()
+
+        # Loop over rates, age, and time
+        for r in rates:
+            df2 = df.loc[df.rate == r].copy()
+
+            ages = np.asarray(sorted(df2.age_lower.unique().tolist()))
+            times = np.asarray(sorted(df2.age_lower.unique().tolist()))
+            n_draws = int(len(df2) / (len(ages) * len(times)))
+
+            # Save these for later for quality checks
+            rate_dict[r]['ages'] = ages
+            rate_dict[r]['times'] = times
+            rate_dict[r]['n_draws'] = n_draws
+
+            # Create template for filling in the draws
+            draw_data = np.zeros((n_draws, len(ages), len(times)))
+            for age_idx, age in enumerate(ages):
+                for time_idx, time in enumerate(times):
+                    # Subset to the draws that we want from avg_integrand
+                    # but only for this particular age and time
+                    draws = df2.loc[
+                        (df2.age_lower == age) &
+                        (df2.time_lower == time)
+                    ]['avg_integrand'].values
+
+                    # Check to makes sure that the number of draws corresponds to the number
+                    # of draws for the whole thing per age and time
+                    assert len(draws) == n_draws
```

Comment:
If we changed this to if not len(draws) == n_draws raise ValueError, or something like that we could catch the time and age for which it failed.  Unless we are already catching this error higher up or something.